tools/splat_ext/pm_player_sprites.py

```diff
@@ -0,0 +1,643 @@
+    def __init__(self, rom_start, rom_end, type, name, vram_start, args, yaml):
+        super().__init__(
+            rom_start, rom_end, type, name, vram_start, args=args, yaml=yaml
+        )
+
+        with (Path(__file__).parent / f"player_sprite_names.yaml").open("r") as f:
+            self.player_cfg = yaml_loader.load(f.read(), Loader=yaml_loader.SafeLoader)
+
+    def out_path(self):
+        return options.opts.asset_path / "sprite/player"
+
+    def split(self, rom_bytes) -> None:
+        player_sprite_cfg = self.player_cfg["player_sprites"]
+        player_raster_names: List[str] = self.player_cfg["player_rasters"]
+
+        player_sprite_names = []
+        for sprite_name in player_sprite_cfg.keys():
+            player_sprite_names.append(sprite_name)
+            if player_sprite_cfg[sprite_name].get("has_back", False):
+                player_sprite_names.append(sprite_name)
+
+        sprite_in_bytes = rom_bytes[self.rom_start : self.rom_end]
+        build_date = sprite_in_bytes[0:0x10].decode("ascii").rstrip("\0")
+        player_raster_offset = int.from_bytes(sprite_in_bytes[0x10:0x14], "big") + 0x10
+        player_yay0_offset = int.from_bytes(sprite_in_bytes[0x14:0x18], "big") + 0x10
+        npc_yay0_offset = int.from_bytes(sprite_in_bytes[0x18:0x1C], "big") + 0x10
+        # sprite_end_offset = int.from_bytes(sprite_in_bytes[0x1C:0x20], "big") + 0x10
+
+        player_sprite_raster_data: bytes = sprite_in_bytes[
+            player_raster_offset:player_yay0_offset
+        ]
+
+        # Header parsing
+        index_ranges_offset = int.from_bytes(player_sprite_raster_data[0:0x4], "big")
+        raster_info_offset = int.from_bytes(player_sprite_raster_data[0x4:0x8], "big")
+        ci4_raster_data_offset = int.from_bytes(
+            player_sprite_raster_data[0x8:0xC], "big"
+        )
+
+        index_ranges = player_sprite_raster_data[index_ranges_offset:raster_info_offset]
+        raster_info = player_sprite_raster_data[
+            raster_info_offset:ci4_raster_data_offset
+        ]
+        # ci4_raster_data = player_sprite_raster_data[ci4_raster_data_offset:]
+
+        # Parse raster sets (readSpriteSections)
+        raster_sets: List[PlayerSpriteRasterSet] = []
+        for i in range(0, len(index_ranges) - 4, 4):
+            start = int.from_bytes(index_ranges[i : i + 4], "big")
+            end = int.from_bytes(index_ranges[i + 4 : i + 8], "big")
+            raster_sets.append(PlayerSpriteRasterSet(start, end - start))
+
+        raster_table_entry_dict = extract_raster_table_entries(raster_info, raster_sets)
+
+        player_sprites = extract_sprites(
+            sprite_in_bytes[player_yay0_offset:npc_yay0_offset], raster_sets
+        )
+
+        #########
+        # Writing
+        #########
+
+        self.out_path().mkdir(parents=True, exist_ok=True)
+        # Write build info (date)
+        with open(self.out_path() / "build_info.txt", "w") as f:
+            f.write(build_date)
```

Comment:
Not sure what's better, but we hardcode this string depending on verson when for building mapfs, and in shiftable mode we use the current date

Reply:
Yeah I don't think it matters, but I like having it extracted

Reply:
Actually, not with the current namescheme. But #1055 makes me want to ask for a more generic name like `sprites.xml` given that its already in the `player/` directory

Reply:
basically what im thinking is like
assets/*/sprite/{npc,player}/sprites.xml
```
<xml>
   <BuildDate>10 AM dog time</BuildDate>
   <Names>
     ...
   </Names>
</xml>
```

Reply:
shounds good